File: app/src/main/java/ai/elimu/analytics/dao/WordLearningEventDao.java
package ai.elimu.analytics.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ai.elimu.analytics.entity.WordLearningEvent;

@Dao
public interface WordLearningEventDao {

    @Insert
    void insert(WordLearningEvent wordLearningEvent);

    @Query("SELECT * FROM WordLearningEvent ORDER BY time DESC")
    List<WordLearningEvent> loadAll();
}

File: app/src/main/java/ai/elimu/analytics/dao/WordLearningEventDao.kt
package ai.elimu.analytics.dao

import ai.elimu.analytics.entity.WordLearningEvent
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface WordLearningEventDao {
    @Insert
    fun insert(wordLearningEvent: WordLearningEvent)

    @Query("SELECT * FROM WordLearningEvent ORDER BY timestamp ASC")
    fun loadAllOrderedByTimestampAsc(): List<WordLearningEvent>

    @Query("SELECT * FROM WordLearningEvent ORDER BY timestamp DESC")
    fun loadAllOrderedByTimestampDesc(): Cursor

    @Query("SELECT COUNT(*) FROM WordLearningEvent")
    fun getCount(): Int
}
